### ISSUE TITLE
Godot freezes permanently when saving any file

### ISSUE THREAD

**Issue by lufog** [godotengine/godot#60148]
### Godot version

v4.0.alpha.custom_build [f62f36cd9]

### System information

Windows 11,  Vulkan Clustered, Geforce 1660 (512.15)

### Issue description

When saving any file, scene or script, Godot freezes permanently.

https://user-images.githubusercontent.com/10379487/162800968-489af082-6349-4ec9-af56-987cae4cafde.mp4

If you reopen the same project, Godot immediately freezes.

### Steps to reproduce

* create empty project;
* add script;
* change it;
* save it.

**Comment by KoBeWi** [comment]
I tried the reproduction steps and now I can't open my project at all. There is empty window and everything is unresponsive:
![image](https://user-images.githubusercontent.com/2223172/162841665-573212dd-343f-4814-825f-03e829ef871e.png)

**Comment by akien-mga** [comment]
I can't reproduce on Linux.

I assume this would be a recent regression (not present in 4.0 alpha 6)?

**Comment by KoBeWi** [comment]
Yes, it's from today. I'm bisecting it right now.

**Comment by KoBeWi** [comment]
Regression from #59440
CC @bruvzg 

I did a bit testing. When I create a new script as described in the OP and save it (or sometimes just after creating it), the project becomes basically corrupted. Removing `.godot` folder doesn't help, the white dialog will always appear at startup and block everything. Only removing the newly created file helps.

But it gets more bizarre though. If the same project is then opened in a build without 9381acb6a42da653cb6dfd9e610dfccead11aa98, the file becomes magically ok and then you can open the project normally in 9381acb6a42da653cb6dfd9e610dfccead11aa98 again. Also if your project is corrupted, it starts normally if you remove the broken file, but the issue returns when you restore the file. Maybe 9381acb6a42da653cb6dfd9e610dfccead11aa98 makes the file saved with some wrong file metadata or something and opening working build fixes it, idk.

**Comment by bruvzg** [comment]
I can't reproduce it on macOS, so it's something Windows specific, probably some of the `FileAccess` that was deleted manually, stay open too long after #59440.

**Comment by bruvzg** [comment]
Seems like there was an issue with preview generator, as I suspected files stay open for too long. I'll check the rest of the changes to make sure the scope of `Ref<FileAccess>` is correct and open PR.

**Comment by SaracenOne** [comment]
@bruvzg Testing again with the latest master, but this does appear have fully resolved this issue. In this case, I get a white box popping up when loading a project which I can't click past

**Comment by SaracenOne** [comment]
[freezes_on_master.zip](https://github.com/godotengine/godot/files/8476632/freezes_on_master.zip)
Here's an example. I think the popup window is probably reporting about a broken path in the main scene, but regardless, you still can't click past it. I'm testing this on an MSVC debug build.

**Comment by SaracenOne** [comment]
Actually, the popup is appearing if the test.tres file is in the project, which is...just a basic StandardMaterial3D. If you take it out of the project, you can bypass the white screen, but it will appear again if you try to create a new StandardMaterial3D resource and save it.

**Comment by SaracenOne** [comment]
@bruvzg I aplogize, misunderstanding. Somehow I stupidly ended up building the wrong branch. I'll attribute some of that to our projecting having a very convoluted automated merging script

**Comment by theVoidZ** [comment]
Still happening sometimes in beta4. Linux.
Freezes at random times after saving, 

and this time, created a scene from inherited scene, Console log looping an error message.
![image](https://user-images.githubusercontent.com/13403351/201033854-2dc9f29c-08b8-4569-ab86-6a6a62c20a8a.png)

**Comment by KoBeWi** [comment]
@theVoidZ Do you maybe have some minimal reproduction project?

**Comment by theVoidZ** [comment]
> @theVoidZ Do you maybe have some minimal reproduction project?

it is not consistent but i ll try to make a new project and try to reproduce it in a separate issue.

**Comment by theVoidZ** [comment]
@KoBeWi 
It is difficult to reproduce. i dont know the exact reasons. 
But If i describe the way it happens sometimes.
i am opening both Godot 3.5 and Godot 4 and reporting/reworking an old project, what i do is navigate between both project and recreates scenes/scripts in Godot 4 as normal. and sometimes while going back to Godot 4, i found it froze and unresponsive. ( have to manually kill the process ). 
Some times i realize it froze after a save ( Ctrl+S ), sometimes when i switch back to Godot 4 window. Sometimes also, when i created an inheritance instance of scene. ( probably after a save of that scene )

**Comment by KoBeWi** [comment]
You could try using a debugger, but if you never compiled Godot it would be difficult.
Other than that, there's little we can do without more info.
